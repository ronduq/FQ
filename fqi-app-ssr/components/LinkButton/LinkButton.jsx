import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Link from 'next/link';
import Icon from '../Icon/Icon';
import t from '../Icon/Icon.module.scss';

import s from './LinkButton.module.scss';

const LinkButton = ({label, href, as}) => {
  return (
    <Link href={href} as={as}>
      <a className={s.linkButton}>
        {label} <Icon className={classNames(s.arrowIcon, t.iconSmall, t.iconOrange, s.arrowRight)} iconName={`chevron`} />
      </a>
    </Link>
  );
};

LinkButton.defaultProps = {
  href: '/',
  as: '/'
};

LinkButton.propTypes = {
  link: PropTypes.string,
  as: PropTypes.string,
};

export default LinkButton;
