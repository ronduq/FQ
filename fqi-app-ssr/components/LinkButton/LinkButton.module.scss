@import '../../style/settings/colours';
@import '../../style/settings/fonts';

a.linkButton {
  color: $colour-secondary;
  cursor: pointer;
  text-transform: uppercase;
  @include font-family('futura');
  @include font-size('medium');

  .arrowRight {
    transform: translateY(2px) rotate(270deg);
    transition: ease-in-out all .1s;
  }

  &:hover {
    color: $colour-secondary;
    text-decoration: underline;
  }
}

