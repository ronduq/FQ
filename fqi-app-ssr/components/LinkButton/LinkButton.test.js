/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import LinkButton from './LinkButton'


describe('With Snapshot Testing', () => {
  it('LinkButton snapshot"', () => {
    const component = renderer.create(<LinkButton />)
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  })
})