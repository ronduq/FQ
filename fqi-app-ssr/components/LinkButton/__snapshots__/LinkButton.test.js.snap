// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing LinkButton snapshot" 1`] = `
<a
  className="linkButton"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
   
  <svg
    aria-hidden="true"
    className="arrowIcon iconSmall iconOrange arrowRight"
    focusable="false"
  >
    <use
      xlinkHref="#chevron"
    />
  </svg>
</a>
`;
